docs/proposals/Periodic-Retrieval-of-Certificates-and-Keys-from-AKV.md

```diff
@@ -0,0 +1,121 @@
+
+### Periodic retrieval of certificates and keys from AKV
+
+To periodically retrieve certificates and keys from AKV, users configure the `KeyManagementProvider` without specifying the version for certificates and keys. Users should have the ability to customize the time interval at which the retrieval process occurs, allowing them to override the default interval (e.g., 24 hours). As part of this enhancement, a new parameter named `retrievalInterval` will be introduced for AKV as the key management provider.
+
+The default value for `retrievalInterval` determines how long the new versions of keys or certificates will be available for verifying images. For Notary Project signatures, it is the root CA certificate retrieved and configured in trust store. In most cases, root CA certificates have a long validity period. Normally certificate rotation will not result in a change on root CA certificate. For Cosign signatures with key pairs in AKV, it depends on how quickly the pipeline switching to use the latest version for signing. The retrieval interval should also consider the impact on normal verification traffic and rotation frequency in AKV. The current recommendation for default value is 24 hours, and users can configure a proper value based on their own situations.
+
+Below is an example of certificate configuration for resource `KeyManagementProvider`. The version for the certificate should not be specified.
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    certificates:
+    - name: ${CERT_NAME}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+```
+
+Below is an example of key configuration for resource `KeyManagementProvider`. The version for the key should not be specified.
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    keys:
+    - name: ${KEY_NAME}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+```
+
+Below is an example to override the default retrieval interval:
+
+```yaml
+apiVersion: config.ratify.deislabs.io/v1beta1
+kind: KeyManagementProvider
+metadata:
+name: keymanagementprovider-akv
+spec:
+type: azurekeyvault
+parameters:
+    vaultURI: https://${AKV_NAME}.vault.azure.net/
+    keys:
+    - name: ${KEY_NAME}
+    tenantID: ${TENANT_ID}
+    clientID: ${IDENTITY_CLIENT_ID}
+    retrievalInterval: 4h
+```
+
+### Periodically reconciling versioned certificates or keys
+
+Users can specify the version for the certificate or key. In this case, reconciliation only takes place for the specified version, rather than retrieving all available versions for the certificate or key.
```

Comment:
I think this behavior is currently inconsistent with what was said on Line 41?

Reply:
This is when users specify the version, which means they want to pin to a specific version. Then the periodic retrieval will only check the status of the specific version, as you mentioned before, it can be disabled for some reasons.